Queries.md

```diff
@@ -0,0 +1,419 @@
+```
+
+
+## Histogram to show the deposited amount of tokens per day
+
+```sql
+with t AS (
+    SELECT generate_series(mitstamp,matstamp,'1 days') AS int
+    FROM 
+    (
+        SELECT min(timestamp) mitstamp, now()::timestamp AS matstamp 
+        FROM "event" 
+        WHERE event = 'ChannelOpened' AND address = '0xa5C9ECf54790334B73E5DfA1ff5668eB425dC474' 
+    ) a
+    order BY int asc
+)
+SELECT timestampwindowstart, COALESCE(sum, 0) AS deposited
+FROM (
+	SELECT 
+		int AS timestampwindowstart,
+		SUM(deposited) filter (WHERE timestamp >= t.int AND timestamp < (t.int + interval '1 days'))
+	FROM
+		t, (
+			SELECT timestamp, (args->2#>'{num}')::numeric AS deposited 
+			FROM "event"
+			WHERE event = 'ChannelNewDeposit' AND address = '0xa5C9ECf54790334B73E5DfA1ff5668eB425dC474'
+		) a
+	GROUP BY
+	  timestampwindowstart
+	order BY
+	  timestampwindowstart
+) res
+```
+
+
+## Max AND Min deposit
+
+```sql
+Select max((args->2#>'{num}')::numeric), min((args->2#>'{num}')::numeric)
+FROM "event"
+WHERE event = 'ChannelNewDeposit' AND address = '0xa5C9ECf54790334B73E5DfA1ff5668eB425dC474';
+```
+
+
+## All participants that no longer have a channel.
+
+In order to get the number of participants that don't have a channel using the given events.
+The following approach was chosen:
+
+`` ALL PARTICIPANTS - PARTICIPANTS WHO STILL HAVE A CHANNEL = *PARTICIPANTS WHO DON'T HAVE A CHANNEL ANYMORE`` 
+
+where ``PARTICIPANTS WHO STILL HAVE A CHANNEL = All participants who are part of the ChannelOpened event and for which the channel was never closed`` 
+
+
+```sql
+with 
+separated_participants AS 
+(
+    SELECT  args->1#>'{hex}' AS participant1, args->2#>'{hex}' AS participant2
+	FROM "event"
+    WHERE event = 'ChannelOpened' AND address = '0xa5C9ECf54790334B73E5DfA1ff5668eB425dC474' 
+), 
+all_participants AS (
+    SELECT participant1 AS participant
+    FROM separated_participants
+    union
+    SELECT participant2 AS participant
+    FROM separated_participants
+), 
+participants_who_have_a_channel_separated AS 
+(
+    SELECT channel_args->1#>'{hex}' participant1, channel_args->2#>'{hex}' participant2
+    FROM
+    (
+        SELECT (args->0#>'{num}')::numeric AS channel_id, args AS channel_args
+        FROM "event"
+        WHERE event = 'ChannelOpened' AND address = '0xa5C9ECf54790334B73E5DfA1ff5668eB425dC474'
+    ) channel_opened_data
+    WHERE channel_id NOT IN 
+    (
+        SELECT (args->0#>'{num}')::numeric AS channel_id
+        FROM "event"
+        WHERE event = 'ChannelClosed' AND address = '0xa5C9ECf54790334B73E5DfA1ff5668eB425dC474' 
+    )
+)
+
+SELECT participant
+FROM all_participants
+WHERE participant NOT IN 
+(
+    SELECT participant1 AS participant 
+    FROM participants_who_have_a_channel_separated 
+    union
+    SELECT participant2 AS participant 
+    FROM participants_who_have_a_channel_separated 
+)
+```
+
+
+## AVG number of channels a node opened
```

Comment:
**AVG number of channels a node opened**
If this could be converted to a date histogram would great.

Reply:
I don't understand how the date histogram will look like... What is the desired output?